src/Avalonia.Themes.Fluent/Controls/ColorSpectrum.xaml

```diff
@@ -0,0 +1,138 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        x:CompileBindings="True"
+        xmlns:converters="using:Avalonia.Controls.Converters">
+
+  <Design.PreviewWith>
+    <Border Padding="20">
+      <ColorSpectrum Width="256" Height="256" />
+    </Border>
+  </Design.PreviewWith>
+
+  <Styles.Resources>
+    <converters:EnumValueEqualsConverter x:Key="EnumValueEquals" />
+  </Styles.Resources>
+
+  <Style Selector="ColorSpectrum">
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate>
+          <Grid x:Name="PART_LayoutRoot">
+            <Grid x:Name="PART_SizingGrid"
+                  HorizontalAlignment="Center"
+                  VerticalAlignment="Center">
+              <Grid.Clip>
+                <RectangleGeometry />
+              </Grid.Clip>
+              <Rectangle x:Name="PART_SpectrumRectangle"
+                         IsHitTestVisible="False"
+                         HorizontalAlignment="Stretch"
+                         VerticalAlignment="Stretch"
+                         IsVisible="{TemplateBinding Shape, Converter={StaticResource EnumValueEquals}, ConverterParameter='Box'}" />
```

Comment:
I think it makes sense to have pseudoclass for ":box" and ":ring" states.
On other hand, I am good with current solution as well.

Reply:
I saw that is what you did in your port. I decided against it for a few reasons:

 1. It is very verbose in my opinion, you had to add a lot of additional style setters for it
 2. It is considerably easier to understand and modify using a converter like I did. There is a lot less code here and much less risk of error for re-templating.
 3. The information required to make these visible/collapsed already exists in a property that is easily accessible. It's not something like selector color/size that is a more complex internal calculation and has no corresponding property. This is simply using the direct value of an existing public property -- a perfect case for just using a converter.